### PR TITLE
Revert "Merge pull request #526 from zendesk/amartinez/revert_grosser"

### PR DESCRIPTION
**Pull request by grosser**
rerevert the kill commit since it was found to be not guilty ...
played with it a bit on staging, but I cannot find a way to make it not work or simulate the remaining stuck errors we see on prod ... should still be a step better then before

@anamartinez @zendesk/runway 

### Risks
 - None

### CODE REVIEW

File: app/models/job_execution.rb [hunk 1]
```diff
@@ -37,7 +38,12 @@ def wait!
   end
 
   def stop!
-    @executor.stop!
+    @executor.stop! 'INT'
+    self.class.stop_timeout.times do
```
Comment (jonmoter) [inline app/models/job_execution.rb]: if `stop_timeout` is a time (like `15.seconds`), you shouldn't be doing `stop_timeout.times`

File: app/models/terminal_executor.rb [hunk 2]
```diff
@@ -33,10 +33,10 @@ def execute!(*commands)
     execute_command!(command)
   end
 
-  def stop!
+  def stop!(signal)
     # Need pkill because we want all
     # children of the parent process dead
-    `pkill -INT -P #{pid}` if pid
+    `pkill -#{signal} -P #{pid}` if pid
```
Comment (jonmoter) [inline app/models/terminal_executor.rb]: We probably should guard against command-line injection attacks here.

File: app/models/job_execution.rb (hunk 1 above)
Comment (grosser) [inline app/models/job_execution.rb]: changed it to .to_i.times to read better ... but works either way

File: app/models/terminal_executor.rb (hunk 2 above)
Comment (grosser) [inline app/models/terminal_executor.rb]: changed to system + array

**Comment by jonmoter** [comment]
:+1:

**Pull request merged**